caffe2/utils/math/reduce.cu

```diff
@@ -533,6 +517,7 @@ DELEGATE_CUDA_REDUCE_FUNCTION(std::int32_t, ReduceSum, cub::Sum, 0)
 DELEGATE_CUDA_REDUCE_FUNCTION(std::int64_t, ReduceSum, cub::Sum, 0LL)
 DELEGATE_CUDA_REDUCE_FUNCTION(float, ReduceSum, cub::Sum, 0.0f)
 DELEGATE_CUDA_REDUCE_FUNCTION(double, ReduceSum, cub::Sum, 0.0)
+DELEGATE_CUDA_REDUCE_FUNCTION(at::Half, ReduceSum, cub::Sum, at::Half(0.0))
```

Comment:
For ReduceSum part, I'd suggest write another specialization function. Since by default for half type computation, we will convert it to float32 and then convert back, while we cannot assume that cub::Sum will do this.